main.tf

```diff
@@ -46,6 +46,16 @@ resource "azurerm_kubernetes_cluster" "main" {
   } /*<box>*/ : replace(k, "avm_", var.tracing_tags_prefix) => v } : {}) /*</box>*/))
   workload_identity_enabled = var.workload_identity_enabled
 
+  dynamic "http_proxy_config" {
+    for_each = var.http_proxy_config == null ? [] : ["http_proxy_config"]
+    content {
+      http_proxy  = var.http_proxy_config.http_proxy
+      https_proxy = var.http_proxy_config.https_proxy
+      no_proxy    = coalesce(var.http_proxy_config.no_proxy, [])
```

Comment:
According to the [`coalesce`'s document](https://developer.hashicorp.com/terraform/language/functions/coalesce), I don't think `[]` is suitable for `coalesce`'s default value. I would recommend `var.http_proxy_config.no_proxy` here, since according to the [schema](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_cluster_resource.go#L447-L453) this field is optional, so a `null` is enough as default value.

Reply:
Done!, thanks for the suggestion